### PR TITLE
cray: add comment about module use in packages.yaml

### PR DESCRIPTION
**Pull request by cwsmith**
Discourage users from putting `paths:` in their `packages.yaml` to avoid problems with the cray programming environment and compiler wrapper.

### CODE REVIEW

File: lib/spack/docs/getting_started.rst [hunk 1]
```diff
@@ -1203,7 +1203,10 @@ Here's an example of an external configuration for cray modules:
 This tells Spack that for whatever package that depends on mpi, load the
 cray-mpich module into the environment. You can then be able to use whatever
 environment variables, libraries, etc, that are brought into the environment
-via module load.
+via module load.  Note, it is best to use `modules:` instead of `paths:` in 
+`packages.yaml` as the Cray Proramming Environment heavily relies on modules
```
Comment (tgamblin) [inline lib/spack/docs/getting_started.rst]: I know it's already in the Cray section but I would reword slightly and put it in a note block.  Also you need ``, not just \` b/c it's `rst` not markdown.

```rst
.. note::

    For Cray-provided packages, it is best to use ``modules:`` instead of ``paths:`` 
    in ``packages.yaml``, because ...

```

Comment (cwsmith) [reply lib/spack/docs/getting_started.rst]: Looks good.  Thank you.
